comptime base64 decode error for long input?

### Zig Version

0.8.1

### Steps to Reproduce

I am compiling down to wasm.  This builds and runs if I only call getPlay(), the other two functions generate the error below.  So I assume it is something to do with the length of the input into the base64 decoder?

const std = @import("std");
const b64 = @import("std").base64.standard;

const PLAY = "PHN2ZyB2ZXJzaW9uPSIxLjEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmlld0JveD0iMCAwIDMyIDMyIiB3aWR0aD0iMzIiIGhlaWdodD0iMzIiPjxyZWN0IGZpbGw9IiMwMDAwMDAiIGZpbGwtb3BhY2l0eT0iMC4zIiB3aWR0aD0iMzIiIGhlaWdodD0iMzIiLz48cGF0aCBzdHJva2U9IiNGQUZBRkEiIHN0cm9rZS13aWR0aD0iMyIgc3Ryb2tlLWxpbmVqb2luPSJyb3VuZCIgc3Ryb2tlLW1pdGVybGltaXQ9IjEwIiBmaWxsPSJub25lIiBkPSJNMTIsMTAgMTIsMjIgMjAsMTYgWiIvPjwvc3ZnPg==";
const PAUSE = "PHN2ZyB2ZXJzaW9uPSIxLjEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmlld0JveD0iMCAwIDMyIDMyIiB3aWR0aD0iMzIiIGhlaWdodD0iMzIiPjxyZWN0IGZpbGw9IiMwMDAwMDAiIGZpbGwtb3BhY2l0eT0iMC4zIiB3aWR0aD0iMzIiIGhlaWdodD0iMzIiLz48cGF0aCBzdHJva2U9IiNGQUZBRkEiIHN0cm9rZS13aWR0aD0iMyIgc3Ryb2tlLWxpbmVjYXA9InJvdW5kIiBzdHJva2UtbGluZWpvaW49InJvdW5kIiBzdHJva2UtbWl0ZXJsaW1pdD0iMTAiIGZpbGw9Im5vbmUiIGQ9Ik0xMiwxMCAxMiwyMiIvPjxwYXRoIHN0cm9rZT0iI0ZBRkFGQSIgc3Ryb2tlLXdpZHRoPSIzIiBzdHJva2UtbGluZWNhcD0icm91bmQiIHN0cm9rZS1saW5lam9pbj0icm91bmQiIHN0cm9rZS1taXRlcmxpbWl0PSIxMCIgZmlsbD0ibm9uZSIgZD0iTTIwLDEwIDIwLDIyIi8+PC9zdmc+";
const CLOSE = "PHN2ZyB2ZXJzaW9uPSIxLjEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmlld0JveD0iMCAwIDMyIDMyIiB3aWR0aD0iMzIiIGhlaWdodD0iMzIiPjxyZWN0IGZpbGw9IiMwMDAwMDAiIGZpbGwtb3BhY2l0eT0iMC4zIiB3aWR0aD0iMzIiIGhlaWdodD0iMzIiLz48cGF0aCBzdHJva2U9IiNGRkZGRkYiIHN0cm9rZS13aWR0aD0iMyIgc3Ryb2tlLWxpbmVjYXA9InJvdW5kIiBzdHJva2UtbWl0ZXJsaW1pdD0iMTAiIGZpbGw9Im5vbmUiIGQ9Ik0yMCwxMiAxMiwyMCIvPjxwYXRoIHN0cm9rZT0iI0ZGRkZGRiIgc3Ryb2tlLXdpZHRoPSIzIiBzdHJva2UtbGluZWNhcD0icm91bmQiIHN0cm9rZS1taXRlcmxpbWl0PSIxMCIgZmlsbD0ibm9uZSIgZD0iTTEyLDEyIDIwLDIwIi8+PC9zdmc+";

const PLAY_DECODED = decodeB64(PLAY);
const PAUSE_DECODED = decodeB64(PAUSE);
const CLOSE_DECODED = decodeB64(CLOSE);

fn decodeB64(comptime encoded: []const u8) [*:0]const u8 { //strange that this has to be comptime for the destLen to be comptime...
    const destLen: usize = b64.Decoder.calcSizeForSlice(encoded) catch unreachable;
    //var buffer: [destLen]u8 = undefined; //looks strange that it is usable but undefined?
    var buffer: [destLen]u8 = [_]u8{0} ** destLen; //maybe this is safer anyway...
    const dest: []u8 = buffer[0..destLen];
    b64.Decoder.decode(dest, encoded) catch unreachable;
    return @ptrCast([*:0]const u8, dest); //is it okay to return a slice onto a comptime array var?
}

pub fn getPlay() [*:0]const u8 {
    return PLAY_DECODED;
}
pub fn getPause() [*:0]const u8 {
    return PAUSE_DECODED;
}
pub fn getClose() [*:0]const u8 {
    return CLOSE_DECODED;
}

### Expected Behavior

no compile error.  
Apologies if this has been fixed in later versions.  I am unable to test beyond 0.8.1 since support for older macos was dropped.

### Actual Behavior

~/Downloads/zig081/lib/std/base64.zig:203:9: error: evaluation exceeded 1000 backwards branches
        for (source) |c, src_idx| {
        ^
./abc.zig:17:23: note: called from here
    b64.Decoder.decode(dest, encoded) catch unreachable;
                      ^
./abc.zig:9:32: note: called from here
const PAUSE_DECODED = decodeB64(PAUSE);
                               ^
./abc.zig:9:32: note: called from here
const PAUSE_DECODED = decodeB64(PAUSE);
                               ^
./abc.zig:17:23: note: referenced here
    b64.Decoder.decode(dest, encoded) catch unreachable;
                      ^
./abc.zig:9:32: note: referenced here
const PAUSE_DECODED = decodeB64(PAUSE);
                               ^
./abc.zig:25:12: note: referenced here
    return PAUSE_DECODED;

--- a/src/Sema.zig
+++ b/src/Sema.zig
@@ -18370,7 +18370,20 @@ fn safetyPanic(
 fn emitBackwardBranch(sema: *Sema, block: *Block, src: LazySrcLoc) !void {
     sema.branch_count += 1;
     if (sema.branch_count > sema.branch_quota) {
-        return sema.fail(block, src, "evaluation exceeded {d} backwards branches", .{sema.branch_quota});
+        const msg = try sema.errMsg(
+            block,
+            src,
+            "evaluation exceeded {d} backwards branches",
+            .{sema.branch_quota},
+        );
+        try sema.errNote(
+            block,
+            src,
+            msg,
+            "use @setEvalBranchQuota() to raise the branch limit from {d}",
+            .{sema.branch_quota},
+        );
+        return sema.failWithOwnedErrorMsg(block, msg);
     }
 }
 

--- a/src/stage1/ir.cpp
+++ b/src/stage1/ir.cpp
@@ -5769,8 +5769,10 @@ static bool ir_emit_backward_branch(IrAnalyze *ira, AstNode* source_node) {
 
     *bbc += 1;
     if (*bbc > *quota) {
-        ir_add_error_node(ira, source_node,
+        ErrorMsg *msg = ir_add_error_node(ira, source_node,
                 buf_sprintf("evaluation exceeded %" ZIG_PRI_usize " backwards branches", *quota));
+        add_error_note(ira->codegen, msg, source_node,
+            buf_sprintf("use @setEvalBranchQuota to raise branch limit from %" ZIG_PRI_usize, *quota));
         return false;
     }
     return true;

--- a/test/cases/recursive_inline_function.1.zig
+++ b/test/cases/recursive_inline_function.1.zig
@@ -14,6 +14,7 @@ inline fn fibonacci(n: usize) usize {
 // error
 //
 // :11:21: error: evaluation exceeded 1000 backwards branches
+// :11:21: note: use @setEvalBranchQuota() to raise the branch limit from 1000
 // :11:40: note: called from here (6 times)
 // :11:21: note: called from here (495 times)
 // :5:24: note: called from here

